lib/index.js

```diff
@@ -119,139 +122,138 @@ function getPolyfillString(options) {
-					const optionsForRUM = cloneDeep(options);
-					Object.keys(options.features).forEach(f => optionsForRUM.features[f].flags.add('always'));
-					return getPolyfills(optionsForRUM)
-						.then(rumFeatures => {
-							return rumTemplate({
-								host: process.env.RUM_BEACON_HOST,
-								features: Object.keys(rumFeatures).reduce((op, featureName) => {
-									const polyfill = sourceslib.getPolyfillMetaSync(featureName);
-									if (polyfill.detectSource) {
-										op.push({name: featureName, detect: polyfill.detectSource});
-									}
-									return op;
-								}, [])
-							});
-						})
-					;
-				});
+		// Using the graph, stream all the polyfill sources in dependency order
+		for (const featureName of sortedFeatures) {
+			const polyfill = sourceslib.getPolyfillMetaSync(featureName);
+			const wrapInDetect = (targetedFeatures[featureName].flags.has('gated') && polyfill.detectSource);
+			if (wrapInDetect) {
+				output.add(streamFromString("if (!(" + polyfill.detectSource + ")) {" + lf));
 			}
 
-			// Using the graph, stream all the polyfill sources in dependency order
-			sortedFeatures.forEach(featureName => {
-				const polyfill = sourceslib.getPolyfillMetaSync(featureName);
-				const wrapInDetect = (targetedFeatures[featureName].flags.has('gated') && polyfill.detectSource);
-				
-				if (wrapInDetect) output.queue("if (!(" + polyfill.detectSource + ")) {" + lf);
-				output.queue(() => {
-					return sourceslib.streamPolyfillSource(featureName, (options.minify ? 'min':'raw'));
-				});
-				if (wrapInDetect) output.queue(lf + "}" + lf + lf);
-			});
-
-			// Invoke the closure, binding `this` to window (in a browser),
-			// self (in a web worker), or global (in Node/IOjs)
-			output.queue("})" + lf + ".call('object' === typeof window && window || 'object' === typeof self && self || 'object' === typeof global && global || {});" + lf);
+			output.add(sourceslib.streamPolyfillSource(featureName, (options.minify ? 'min' : 'raw')));
 
-			if ('all' in options.features) {
-				output.queue("\nconsole.log('"+allWarnText+"');\n");
+			if (wrapInDetect) {
+				output.add(streamFromString(lf + "}" + lf + lf));
 			}
-					
-			output.finalize();
+		}
+		// Invoke the closure, binding `this` to window (in a browser),
+		// self (in a web worker), or global (in Node/IOjs)
+		output.add(streamFromString("})" + lf + ".call('object' === typeof window && window || 'object' === typeof self && self || 'object' === typeof global && global || {});" + lf));
 
-		})
-		.catch(function(err) {
-			console.log(err.stack || err);
-		})
-	;
+		if ('all' in options.features) {
+			output.add(streamFromString("\nconsole.log('"+allWarnText+"');\n"));
+		}
+	});
 
-	return options.stream ? output : new Promise(resolve => {
-		const bufs = [];
-		output.on('data', chunk => bufs.push(chunk));
-		output.on('end', () => resolve(Buffer.concat(bufs).toString('UTF-8')));
+	return options.stream ? output : streamToPromise(output).then(bufs => {
```

Comment:
You could use a higher level lib here like stream-to-string: https://www.npmjs.com/package/stream-to-string